aten/src/ATen/native/vulkan/VulkanCommon.h

```diff
@@ -54,6 +54,21 @@ struct ContextConv2D final {
   static constexpr float kMax = std::numeric_limits<float>::infinity();
 };
 
+namespace detail {
+template <typename To, typename From>
+inline constexpr To safe_downcast(From v) {
```

Comment:
I'm not sure this can be  constexpr anymore considering that some implementations of TORCH_CHECK might throw so if you run into any compilation issues, that's a potential reason.

Reply:
Also please mark `v` as const.